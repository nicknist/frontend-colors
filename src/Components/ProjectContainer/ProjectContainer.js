import React from 'react';
import Project from '../Project/Project';

const ProjectContainer = ({ projects }) => {
  return (
    <>
      {projects.map(project => <Project key={project.title} info={project}/>)}
    </>
  )
}

export default ProjectContainer;
