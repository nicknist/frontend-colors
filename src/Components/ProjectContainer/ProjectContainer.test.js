import React from 'react';
import ProjectContainer from './ProjectContainer';
import { shallow } from 'enzyme';

describe('ProjectContainer', () => {
  it('should match the snapshot', () => {
    const mockProjects = [
      { name: 'PROJECT 1',
        palettes: [
        { name: 'NUMBAH1',
          colors: [
            { color: 'white', locked: false },
            { color: 'blue', locked: false },
            { color: 'red', locked: false },
            { color: 'orange', locked: false },
            { color: 'rebeccapurple', locked: false }
          ]
        }
      ]
      }
    ]
    const wrapper = shallow(<ProjectContainer projects={mockProjects}/>)
    expect(wrapper).toMatchSnapshot();
  })
})
