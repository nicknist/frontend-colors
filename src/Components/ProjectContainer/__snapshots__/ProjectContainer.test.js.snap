// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snapshot 1`] = `
<Fragment>
  <Project
    info={
      Object {
        "name": "PROJECT 1",
        "palettes": Array [
          Object {
            "colors": Array [
              Object {
                "color": "white",
                "locked": false,
              },
              Object {
                "color": "blue",
                "locked": false,
              },
              Object {
                "color": "red",
                "locked": false,
              },
              Object {
                "color": "orange",
                "locked": false,
              },
              Object {
                "color": "rebeccapurple",
                "locked": false,
              },
            ],
            "name": "NUMBAH1",
          },
        ],
      }
    }
  />
</Fragment>
`;
